src/Uno.UI/UI/Xaml/Controls/Frame/Frame.cs

```diff
@@ -432,6 +430,17 @@ private void ReleasePages(IList<PageStackEntry> pageStackEntries)
 
 		public void SetNavigationState(string navigationState) => _navigationState = navigationState;
 
+		internal Page EnsurePageInitialized(PageStackEntry entry)
+		{
+			if (entry is { Instance: null } && CreatePageInstanceCached(entry.SourcePageType) is { } page)
```

Comment:
Should this validation be separate? Like

```csharp
if (entry is { Instance: null })
{
    entry.Instance = page;
}

if(CreatePageInstanceCached(entry.SourcePageType) is { } page) 
{
    page.Frame = this;
}
```

Reply:
I think the current logic is correct - as it needs to initialize the page first and then set the instance